compiler-rs/clients_schema_to_openapi/src/paths.rs

```diff
@@ -251,6 +251,43 @@ pub fn add_endpoint(
 
         let sum_desc = split_summary_desc(&endpoint.description);
 
+        // add the x-state extension for availability
+        let mut extensions = crate::availability_as_extensions(&endpoint.availability);
+
+        // add the x-codeSamples extension
+        if !request_examples.is_empty() {
+            let mut code_samples = vec![];
+            if let Some((_, first_example)) = request_examples.clone().first() {
+                if let Some(example) = first_example.as_item() {
+                    if let Some(description) = example.description.clone() {
+                        let mut request: Option<String> = Option::None;
+                        for r in description.split('`') {
+                            if r.starts_with("GET ")
+                                || r.starts_with("POST ")
+                                || r.starts_with("PUT ")
+                                || r.starts_with("DELETE ")
+                                || r.starts_with("HEAD ")
+                            {
+                                request = Some(String::from(r));
+                                break;
+                            }
```

Comment:
Do we have examples that don't start with a method? QUERY will be a thing soon!

Reply:
This is really a hack. I hope we don't have to do this. The problem is that the examples that were added to this repository are not clean Dev Console examples, so I'm forced to reconstruct the DevConsole syntax.